packages/backend/src/cache/data-stores/redis.test.ts

```diff
@@ -0,0 +1,60 @@
+import { createRedisDataStore } from './redis'
+import Redis from 'ioredis'
+import { Config } from '../../config/app'
+
+describe('Redis Data Store', (): void => {
+  const redis = new Redis(Config.redisUrl, {
+    tls: Config.redisTls,
+    stringNumbers: true
+  })
+
+  const ttlMs = 100
+  const dataStore = createRedisDataStore(redis, ttlMs)
+
+  afterEach(async () => {
+    await redis.flushall()
+  })
+
+  afterAll(async () => {
+    redis.disconnect()
+  })
+
+  describe('set', (): void => {
+    test('returns true if key set properly', async () => {
+      await expect(dataStore.set('foo', 'bar')).resolves.toBe(true)
+    })
+
+    test('keys expire properly', async () => {
+      await expect(dataStore.set('foo', 'bar')).resolves.toBe(true)
+      await expect(dataStore.get('foo')).resolves.toBe('bar')
+      await new Promise((resolve) => setTimeout(resolve, ttlMs))
+      await expect(dataStore.get('foo')).resolves.toBeUndefined()
+    })
+
+    test('values can be overriden', async () => {
+      await expect(dataStore.set('foo', 'bar')).resolves.toBe(true)
+      await expect(dataStore.set('foo', 'barbar')).resolves.toBe(true)
+      await expect(dataStore.get('foo')).resolves.toBe('barbar')
+    })
+  })
+
+  describe('get', (): void => {
+    test('returns undefined if key not set', async () => {
+      await expect(dataStore.get('foo')).resolves.toBeUndefined()
+    })
+    test('returns value if key set', async () => {
+      await expect(dataStore.set('foo', 'bar')).resolves.toBe(true)
+      await expect(dataStore.get('foo')).resolves.toBe('bar')
+    })
+  })
+
+  describe('delete', (): void => {
+    test('returns false if no key deleted', async () => {
+      await expect(dataStore.delete('foo')).resolves.toBe(false)
+    })
```

Comment:
I'm not sure if
> return true if it's trying to delete a key that didn't exist
is intuitive. 

Receiving "true" on an operation signifies to me that the operation was successful, in this case, deleting a key. 

Returning true if item was deleted, false if no key deleted is the behaviour of the javascript operations as well: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete#return_value

Reply:
I think we used to do that for access token revocation but it looks like we don't do that anymore. I just think about the outcome here. If we try to delete a key that does exist and it fails, i.e. you get a `false` back, you can retry deleting. However, if the key doesn't exist and you get back a `false`, you don't want to retry but you can't distinguish between those two cases. That is why I'd return `true` if the key was successfully deleted or doesn't exist, because there is no further action required.

Reply:
Тhat makes sense, but in that case for the redis implementation, the `del` command would need to be followed by another `get` to confirm that the key isn't actually there, regardless of the number of keys that were deleted in the `del` command (i.e. the integer reply): https://redis.io/commands/del/

Reply:
The redis `del` returns true or false, not numbers?

Reply:
It returns:
> [Integer reply](https://redis.io/docs/reference/protocol-spec#resp-integers): The number of keys that were removed.

Reply:
If it successfully deletes one, it returns 1. 
If we try to delete one that does not exist, it returns 0.
What happens when deletion fails? Does it return an error or -1? I feel like I've seen that before (the -1) but I'm don't remember in which context and I can't find an answer in the docs.

Reply:
The only failure I can think of is that it just can't connect to redis itself. I can try and see what happens in that case, maybe it throws instead

Reply:
Basically, the failure that seems realistic that I can think of is a network failure, or if redis is disconnected etc.
In that case, the requests to redis will throw an error -> I can add some try-catch around the commands & add a timeout for `set` `get` `delete`. Thoughts?

Reply:
Just my two cents... the redis docs suggest it returns a 0 if the key doesnt exist - guess it would probably be good to confirm how `ioredis` behaves.

At face value I agree with both:
- `dataStore.delete` should return true if the key exists and was deleted and false if not
- it should not return false for a key that is not found

I guess one more option is throwing an error in `dataStore.delete` if it doesn't exist, otherwise return true/false. Either way it does seem like it will take the extra `get` when the delete response from redis is 0 like max described.

Reply:
>  I can add some try-catch around the commands & add a timeout for set get delete. Thoughts?

Since we never have a try-catch for db mutations, I don't see why we should make an exception here. 
But does that mean that we just always return true on deletion?

Reply:
> But does that mean that we just always return true on deletion?

Maybe in this case, we can just make it a `void` method for now... 😼 

> it should not return false for a key that is not found

This could also be the responsibility of the caller of the data store if (for any reason) they'd like to verify the key does not exist prior to or after deletion

Reply:
Turned delete into a `void` method for the time being, [commit link](https://github.com/interledger/rafiki/pull/1352/commits/dd918ce1037017150966fb38f6208c1671c060a5)